### ISSUE TITLE
Segmentation fault in picoquic_cnx_is_still_logging

### ISSUE THREAD

**Issue by martinthomson** [private-octopus/picoquic#1049]
It looks like enabling logging can result in crashes after the idle timeout expires.

```
Program received signal SIGSEGV, Segmentation fault.
picoquic_cnx_is_still_logging (cnx=0x0) at picoquic/quicctx.c:675
675         int ret =
(gdb) bt
#0  picoquic_cnx_is_still_logging (cnx=0x0) at picoquic/quicctx.c:675
#1  0x000055555557f757 in picoquic_incoming_segment (quic=quic@entry=0x5555555da6b0,
    bytes=bytes@entry=0x7fffffffd160 "...", length=length@entry=29, packet_length=packet_length@entry=29,
    consumed=consumed@entry=0x7fffffffcde8, addr_from=addr_from@entry=0x7fffffffcf60, addr_to=0x7fffffffcfe0, if_index_to=1,
    received_ecn=0 '\000', current_time=1601602575150070, receive_time=1601602575150070, previous_dest_id=0x7fffffffcdf0)
    at picoquic/packet.c:1905
#2  0x0000555555580470 in picoquic_incoming_packet (quic=quic@entry=0x5555555da6b0,
    bytes=bytes@entry=0x7fffffffd160 "...", packet_length=packet_length@entry=29,
    addr_from=addr_from@entry=0x7fffffffcf60, addr_to=addr_to@entry=0x7fffffffcfe0, if_index_to=1, received_ecn=0 '\000',
    current_time=1601602575150070) at picoquic/packet.c:2116
#3  0x00005555555906ed in picoquic_packet_loop (quic=0x5555555da6b0, local_port=<optimized out>, local_af=<optimized out>,
    dest_if=-1, loop_callback=0x55555556a8c0 <server_loop_cb>, loop_callback_ctx=0x7fffffffddf4) at picoquic/sockloop.c:251
#4  0x000055555556c115 in quic_server (server_name=<optimized out>, server_port=4443, pem_cert=<optimized out>,
    pem_key=0x7fffffffe1a0 "./certs/key.pem", just_once=<optimized out>, do_retry=0, cnx_id_callback=0x0,
    cnx_id_callback_ctx=0x0, reset_seed=0x0, dest_if=-1, mtu_max=0, proposed_version=0, esni_key_file_name=0x0,
    esni_rr_file_name=0x0, log_file=0x7fffffffe78e "test.log", bin_dir=0x0, qlog_dir=0x0, use_long_log=0,
    cc_algorithm=<optimized out>, web_folder=0x0, initial_random=0) at picoquicfirst/picoquicdemo.c:219
```
This is the result of running neqo's keyupdate test with broken code (which causes neqo to hit the idle timeout after picoquic).  I think that what is happening is that a packet is arriving after the idle timeout has fired.  Not exactly sure.

Try `picoquicdemo -l blah` and the client of neqo@6c89b13b068134a1012fd9bdb04e1ad58abe319d, which has a bug that will cause this odd situation fairly reliably:

```
git clone https://github.com/mozilla/neqo
cd neqo
git checkout 6c89b13b0
cargo run --bin neqo-client -- --key-update https://[::1]:4443/
```

**Comment by huitema** [comment]
Thanks for filing this, Martin. Teaches me to not "quickly hack packet number randomization..."

**Comment by huitema** [comment]
@martinthomson Fix ready, PR #1050.

The "still logging" function replaces a previous direct test of the sequence number, which was not possible if PN numbers are randomized. But the edit dropped a test of the validity of the connection context.
